lib/ex_plasma/transaction.ex

```diff
@@ -206,103 +287,97 @@ defmodule ExPlasma.Transaction do
 
   @doc """
-  Sign the inputs of the transaction with the given keys in the corresponding order.
+  Signs the inputs of the transaction with the given keys in the corresponding order.
+  Only signs transactions that implement the ExPlasma.TypedData protocol.
 
+  Returns
+  - {:ok, %Transaction{}} with sigs when succesfuly signed
+  or
+  - {:error, :not_signable} when the given transaction is not supported.
 
   ## Example
 
-    iex> key = "0x79298b0292bbfa9b15705c56b6133201c62b798f102d7d096d31d7637f9b2382"
-    iex> txn = %ExPlasma.Transaction{tx_type: 1}
-    iex> ExPlasma.Transaction.sign(txn, keys: [key])
-    %ExPlasma.Transaction{
-        inputs: [],
-        metadata: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-          0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
-        outputs: [],
-        sigs: [
-              <<129, 213, 32, 15, 183, 218, 255, 22, 82, 95, 22, 86, 103, 227, 92, 109, 9,
-                89, 7, 142, 235, 107, 203, 29, 20, 231, 91, 168, 255, 119, 204, 239, 44,
-                125, 76, 109, 200, 196, 204, 230, 224, 241, 84, 75, 9, 3, 160, 177, 37,
-                181, 174, 98, 51, 15, 136, 235, 47, 96, 15, 209, 45, 85, 153, 2, 28>>
-            ],
-        tx_data: 0,
-        tx_type: 1
-    }
+  iex> key = "0x79298b0292bbfa9b15705c56b6133201c62b798f102d7d096d31d7637f9b2382"
+  iex> txn = %ExPlasma.Transaction{tx_type: 1}
+  iex> ExPlasma.Transaction.sign(txn, [key])
+  {:ok, %ExPlasma.Transaction{
+    sigs: [
+          <<129, 213, 32, 15, 183, 218, 255, 22, 82, 95, 22, 86, 103, 227, 92, 109, 9,
+            89, 7, 142, 235, 107, 203, 29, 20, 231, 91, 168, 255, 119, 204, 239, 44,
+            125, 76, 109, 200, 196, 204, 230, 224, 241, 84, 75, 9, 3, 160, 177, 37,
+            181, 174, 98, 51, 15, 136, 235, 47, 96, 15, 209, 45, 85, 153, 2, 28>>
+        ],
+    inputs: [],
+    metadata: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
+    outputs: [],
+    tx_type: 1,
+    witnesses: [],
+    tx_data: 0
+  }}
   """
-  def sign(%{} = transaction, keys: []), do: %{transaction | sigs: []}
-
-  def sign(%{} = transaction, keys: keys) when is_list(keys) do
-    eip712_hash = ExPlasma.TypedData.hash(transaction)
-    sigs = Enum.map(keys, fn key -> ExPlasma.Signature.signature_digest(eip712_hash, key) end)
-    %{transaction | sigs: sigs}
+  def sign(transaction, keys) do
+    case Signed.compute_signatures(transaction, keys) do
+      {:ok, sigs} -> {:ok, Map.put(transaction, :sigs, sigs)}
+      {:error, :not_signable} = error -> error
+    end
   end
 
-  @doc """
-  Keccak hash the Transaction. This is used in the contracts and events to to reference transactions.
-
+  defp parse_tx_type(tx_type_rlp) do
+    with {:ok, tx_type} <- RlpDecoder.parse_uint256(tx_type_rlp),
+         {:ok, module} <- get_transaction_module(tx_type) do
+      {:ok, tx_type, module}
+    else
+      _ -> {:error, :unrecognized_transaction_type}
```

Comment:
is this perhaps rewriting too much?

Reply:
I meant rewriting the error response and perhaps hiding the real error.

Reply:
These https://github.com/omgnetwork/ex_plasma/pull/67/files#diff-1daebe1f30daea2ad1ff05dc2e84c3a6R22 are the errors it's hiding, do you think it's worth forwarding them?
I like to have the simplification here but am open to suggestions!
I mean, with the docs and stuff around how to build a transaction, if you get the error `unrecognized_transaction_type` you should quickly figure out what's wrong as the expected value is very simple (`1`, `3`, ...)